.github/workflows/validate-kind.yaml

```diff
@@ -0,0 +1,121 @@
+name: validate-kind-e2e
+
+on:
+  workflow_dispatch:
+  # pull_request:
+  #   types:
+  #     - opened
+  #     - reopened
+  #   branches:
+  #     - 'renovate/**'
+  push:
+    branches:
+      - feature/add_e2e_kind_action
+
+jobs:
+  kubernetes:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@main
+      - name: Init Homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+      - name: Install Flux
+        run: |
+          brew install fluxcd/tap/flux@2.3
+          flux -v
+      - name: Init Kind action
+        uses: helm/kind-action@v1.12.0
+        with:
+          cluster_name: sqnc-flux-infra
+          config: ./scripts/kind-cluster-config.yaml
+      - name: Install Flux to cluster
+        run: flux install
+      - name: Configure Git
+        run: |
+          git config user.name ${GITHUB_ACTOR}
+          git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+      - name: Patch GitOps branch to match PR
+        if: |
+          github.event_name == 'pull_request' &&
+          contains('refs/heads/renovate', github.ref)
+        run: |
+          patch=${GITHUB_REF#refs/heads/}
+          sed -i -r "s#(branch:).*#\1 $patch#" \
+            ./clusters/kind-cluster/base/flux-system/gotk-sync.yaml
+          git add ./clusters/kind-cluster/base/flux-system/gotk-sync.yaml
+          git commit -am "Patch GitOps branch to match PR"
+          git push -u origin $patch
+      - name: Source repository and reconcile
+        if: success()
+        run: |
+          flux create source git flux-system \
+          --url=${{ github.event.repository.html_url }} \
+          --branch=${GITHUB_REF#refs/heads/} \
+          --username=${GITHUB_ACTOR} \
+          --password=${{ secrets.GITHUB_TOKEN }}
+          flux create kustomization flux-system \
+          --source=flux-system \
+          --path=./clusters/kind-cluster/base
```

Comment:
So what we want to do in these steps is firstly, run against main and ensure everything successfully installs there.

Then once we have successfully got a fully installed kind cluster, we want to change the git source to be the target branch the PR is running against, reconcile and ensure the affected charts upgrade.